homeassistant/components/recorder/models.py

```diff
@@ -222,7 +227,10 @@ class States(Base):  # type: ignore[misc,valid-type]
     attributes_id = Column(
         Integer, ForeignKey("state_attributes.attributes_id"), index=True
     )
-    event = relationship("Events", uselist=False)
+    context_id = Column(String(MAX_LENGTH_EVENT_CONTEXT_ID), index=True)
+    context_user_id = Column(String(MAX_LENGTH_EVENT_CONTEXT_ID))
+    context_parent_id = Column(String(MAX_LENGTH_EVENT_CONTEXT_ID))
+    origin_idx = Column(Integer)  # 0 is local, 1 is remote
```

Comment:
There won't be many origins, can we use [`SmallInteger`](https://docs.sqlalchemy.org/en/14/core/type_basics.html#sqlalchemy.types.SmallInteger) instead?
```suggestion
    origin_idx = Column(SmallInteger)  # 0 is local, 1 is remote
```

Reply:
Yes.  I don't think sqlite or mysql will store if any differently though (maybe postgres does)